cmd/scale.go

```diff
@@ -0,0 +1,271 @@
+	f.IntVar(&sc.newDesiredAgentCount, "new-agent-count", 0, "desired number of agents")
+	f.BoolVar(&sc.classicMode, "classic-mode", false, "enable classic parameters and outputs")
+	f.StringVar(&sc.agentPoolToScale, "agent-pool", "", "agent pool to scale")
+
+	addAuthFlags(&sc.authArgs, f)
+
+	return scaleCmd
+}
+
+func (sc *scaleCmd) validate(cmd *cobra.Command, args []string) {
+	log.Infoln("validating...")
+	var err error
+
+	sc.locale, err = i18n.LoadTranslations()
+	if err != nil {
+		log.Fatalf("error loading translation files: %s", err.Error())
+	}
+
+	if sc.resourceGroupName == "" {
+		cmd.Usage()
+		log.Fatal("--resource-group must be specified")
+	}
+
+	if sc.location == "" {
+		cmd.Usage()
+		log.Fatal("--location must be specified")
+	}
+
+	if sc.newDesiredAgentCount == 0 {
+		cmd.Usage()
+		log.Fatal("--agent-count must be specified")
+	}
+
+	if sc.client, err = sc.authArgs.getClient(); err != nil {
+		log.Error("Failed to get client:", err)
+	}
+
+	if sc.deploymentDirectory == "" {
+		cmd.Usage()
+		log.Fatal("--deployment-dir must be specified")
+	}
+
+	_, err = sc.client.EnsureResourceGroup(sc.resourceGroupName, sc.location, nil)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// load apimodel from the deployment directory
+	sc.apiModelPath = path.Join(sc.deploymentDirectory, "apimodel.json")
+
+	if _, err = os.Stat(sc.apiModelPath); os.IsNotExist(err) {
+		log.Fatalf("specified api model does not exist (%s)", sc.apiModelPath)
+	}
+
+	apiloader := &api.Apiloader{
+		Translator: &i18n.Translator{
+			Locale: sc.locale,
+		},
+	}
+	sc.containerService, sc.apiVersion, err = apiloader.LoadContainerServiceFromFile(sc.apiModelPath, true, nil)
+	if err != nil {
+		log.Fatalf("error parsing the api model: %s", err.Error())
+	}
+
+	if sc.agentPoolToScale == "" {
+		agentPoolCount := len(sc.containerService.Properties.AgentPoolProfiles)
+		if agentPoolCount > 1 {
+			log.Fatal("--agent-pool is required if more than one agent pool is defined in the container service")
+		} else if agentPoolCount == 1 {
+			sc.agentPool = sc.containerService.Properties.AgentPoolProfiles[0]
+			sc.agentPoolIndex = 0
+			sc.agentPoolToScale = sc.containerService.Properties.AgentPoolProfiles[0].Name
+		} else {
+			log.Fatal("No agent pools found to scale")
+		}
+	} else {
+		agentPoolIndex := -1
+		for i, pool := range sc.containerService.Properties.AgentPoolProfiles {
+			if pool.Name == sc.agentPoolToScale {
+				agentPoolIndex = i
+				sc.agentPool = pool
+				sc.agentPoolIndex = i
+			}
+		}
+		if agentPoolIndex == -1 {
+			log.Fatalf("Agent pool %s wasn't in the deployed api model", sc.agentPoolToScale)
+		}
+	}
+}
+
+func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
+	sc.validate(cmd, args)
+
+	if sc.agentPool.IsAvailabilitySets() && sc.agentPool.Count > sc.newDesiredAgentCount {
+		// TODO add scale down of VMS code path
+		log.Fatalln("Scaling down availibility sets is not currently supported")
```

Comment:
Looks like there is a typo in the word `availability`

Reply:
thanks I fixed it